example_unrolling_service/example_without_bazel.py

```diff
@@ -0,0 +1,190 @@
+    to compute incremental reward.
+    """
+
+    def __init__(self):
+        super().__init__(
+            id="size",  # name="size", #FIXME
+            observation_spaces=["size"],
+            default_value=0,
+            default_negates_returns=True,
+            deterministic=False,
+            platform_dependent=True,
+        )
+        self.baseline_size = 0
+
+    def reset(self, benchmark: str, observation_view):
+        del benchmark  # unused
+        self.baseline_runtime = observation_view["size"]
+
+    def update(self, action, observations, observation_view):
+        del action  # unused
+        del observation_view  # unused
+        return float(self.baseline_size - observations[0]) / self.baseline_size
+
+
+class UnrollingDataset(Dataset):
+    def __init__(self, *args, **kwargs):
+        super().__init__(
+            name="benchmark://unrolling-v0",
+            license="MIT",
+            description="Unrolling example dataset",
+        )
+
+        self._benchmarks = {
+            "benchmark://unrolling-v0/offsets1": Benchmark.from_file_contents(  # FIXME: "/offsets1": Benchmark.from_file_contents(
+                "benchmark://unrolling-v0/offsets1",
+                # FIXME: self.preprocess(BENCHMARKS_PATH / "offsets1.c"),
+                self.preprocess(os.path.join(
+                    BENCHMARKS_PATH, "offsets1.c")),  # FIXME: why did we have to add "bytes(..., "utf8")" conversion
+            ),
+            "benchmark://unrolling-v0/conv2d": Benchmark.from_file_contents(  # FIXME: "/conv2d": Benchmark.from_file_contents(
+                "benchmark://unrolling-v0/conv2d",
+                # FIXME: self.preprocess(BENCHMARKS_PATH / "conv2d.c"),
+                self.preprocess(os.path.join(
+                    BENCHMARKS_PATH, "conv2d.c")),  # FIXME: why did we have to add "bytes(..., "utf8")" conversion
+            ),
+        }
+
+    @staticmethod
+    def preprocess(src: Path) -> bytes:
+        """Front a C source through the compiler frontend."""
+        # TODO(github.com/facebookresearch/CompilerGym/issues/325): We can skip
+        # this pre-processing, or do it on the service side, once support for
+        # multi-file benchmarks lands.
+        cmd = [
+            "clang",  # str(llvm.clang_path()), # FIXME
+            "-E",
+            "-o",
+            "-",
+            "-I",
+            str(NEURO_VECTORIZER_HEADER.parent),
+            str(src),
+        ]
+        # cmd += get_system_library_flags() # FIXME
+        for directory in get_system_includes():
+            cmd += ["-isystem", str(directory)]
+        return subprocess.check_output(  # run_command(
+            cmd,
+            timeout=300,
+        )
+
+    def benchmark_uris(self) -> Iterable[str]:
+        # FIXME: yield from (f"benchmark://unrolling-v0{k}" for k in self._benchmarks.keys())
+        yield from self._benchmarks.keys()
+
+    def benchmark(self, uri: str) -> Benchmark:
```

Comment:
Rename `benchmark_from_parsed_uris` to fix this

Reply:
Thanks Chris. It fixed it, but then got another error down the line:

```
Traceback (most recent call last):
  File "example_unrolling_service/example_without_bazel.py", line 171, in <module>
    observation = env.reset()
  File "/Users/melhoushi/opt/anaconda3/envs/compiler_gym_examples/lib/python3.8/site-packages/compiler_gym/envs/compiler_env.py", line 812, in reset
    self._benchmark_in_use_proto.uri = str(self._benchmark_in_use.uri)
AttributeError: 'generator' object has no attribute 'uri'
```